### PR TITLE
chore(enterprise): Add enterprise soak test

### PR DESCRIPTION
**Pull request by 001wwang**
Ref OP-502

This PR
- Adds a soak test to track the performance overhead of enterprise reporting components. This prepares us to change the default scrape interval in enterprise `internal_metrics`.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
I think this soak test will currently fail to actually send metrics to Datadog due to lack of valid enterprise configuration. Is that expected and fine for your purposes? We should probably note it in the README if so as I could see others being confused.

**Comment by 001wwang** [comment]
> I think this soak test will currently fail to actually send metrics to Datadog due to lack of valid enterprise configuration. Is that expected and fine for your purposes? We should probably note it in the README if so as I could see others being confused.

Correct, it won't send anything to Datadog and this is expected. I was originally going to try and set this up to actually send to Datadog, but after chatting with @blt, I realized it's not terribly important as long as the telemetry is sent _somewhere_ (`lading` http blackhole) and it's a lot simpler this way (no need to manage secrets, wire up env variables to the soak scripts, handle flaky issues with DD endpoints, etc.).

I'll add these details to the README.

File: soaks/tests/enterprise_http_to_http/vector.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+data_dir = "/var/lib/vector"
+
+##
+## Enterprise
+##
+[enterprise]
+api_key = "DD_API_KEY"
+application_key = "DD_APPLICATION_KEY"
+configuration_key = "DD_CONFIGURATION_KEY"
+endpoint = "http://localhost:8080"
```
Comment (jszwedko) [inline soaks/tests/enterprise_http_to_http/vector.toml:10]: Aha, I missed this `endpoint` configuration. 👍

**Review by jszwedko — APPROVED** [review]
Looks good! Thanks for the additional explanation.

**Pull request merged**